lesson7/thingspeak_cpu_loop.py

```diff
@@ -2,12 +2,19 @@
 from time import localtime, strftime
 import psutil
 import time
+from pickle import dump as pickle_dump
+from pickle import load as pickle_load
+from os.path import exists as file_exists
+from os.path import isfile as is_file
+PICKLE_FILE_NAME = "API_KEY.pickle"
+API_KEY_INDEX = "API_KEY"
+KEY = None
 def doit():
 	cpu_pc = psutil.cpu_percent()
 	mem = psutil.virtual_memory()
 	mem_avail_mb = mem.available/(1024*1024)
 	params = urllib.parse.urlencode({'field1': cpu_pc, 'field2':
-             mem_avail_mb,'key':'YOURKEYHERE'})
+             mem_avail_mb,'key':KEY})
```

Comment:
Removed key storage in source, which is generally a bad practice, as it leads to version control systems having your key.